src/main/java/com/mentorship/vineservice/services/imp/VineServiceImpl.java

```diff
@@ -51,4 +52,19 @@ public VinesDto getVinesWithFilterAndPagination(VinesQueryParameters vinesQueryP
             .build();
     }
 
+    @Override
+    public Vine getVineById(Long vineId) {
+        return vineRepository.findById(vineId)
+            .orElseThrow(() -> new NoSuchElementException(String.format("Vine with id %s not found", vineId)));
+    }
+
+    @Override
+    public void updateVineAmount(VineDto vineDto) {
+
+        Vine vine = getVineById(vineDto.getId());
+        vineMapper.updateVineFromDto(vineDto, vine);
+        vineRepository.save(vine);
```

Comment:
why we update wine amount in such a way?

Reply:
why not?)
Approach with vineMapper.updateVineFromDto allows us for a partial update, I mean update only field that we provide, it will not put null for all other fields